crates/matrix-sdk/src/media.rs

```diff
@@ -96,6 +113,43 @@ impl Media {
         Ok(self.client.send(request, Some(request_config)).await?)
     }
 
+    /// Gets a media file by copying it to a temporary location on disk.
+    ///
+    /// If the content is encrypted and encryption is enabled, the content will
+    /// be decrypted.
+    ///
+    /// Returns a `MediaFileHandle` which takes ownership of the file. When the
+    /// handle is dropped, the file will be deleted from the temporary location.
+    ///
+    /// # Arguments
+    ///
+    /// * `request` - The `MediaRequest` of the content.
+    ///
+    /// * `content_type` - The type of the media, this will be used to set the
+    ///   temporary file's extension.
+    ///
+    /// * `use_cache` - If we should use the media cache for this request.
+    pub async fn get_media_file(
```

Comment:
The naming of this may conflict with `get_file` somewhat (which doesn't return a file). Happy to update if there are better suggestions.